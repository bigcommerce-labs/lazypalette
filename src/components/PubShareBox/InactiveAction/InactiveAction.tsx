import React, {SFC} from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';

interface InactiveActionProps {
    canPublish: boolean;
    canSave: boolean;
    isPrelaunchStore: boolean;
    handlePublish(): void;
    handleSave(): void;
}

const InactiveAction: SFC<InactiveActionProps> = props => {
    const publishTestId = props.isPrelaunchStore ? 'use-as-active' : 'publish';
    const publishActionText = props.isPrelaunchStore ? 'Use as Active Theme' : 'Publish';

    return (
        <>
            <ButtonInput
                onClick={props.handleSave}
                type="button"
                testId="save"
                disabled={!props.canSave}
            >
                Save
            </ButtonInput>
            <ButtonInput
                onClick={props.handlePublish}
                classType="primary"
                disabled={!props.canPublish}
                testId={publishTestId}
            >
                {publishActionText}
            </ButtonInput>
        </>
    );
};

export default InactiveAction;
