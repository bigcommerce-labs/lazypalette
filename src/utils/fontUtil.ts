export function parseFont(font: string) {
    const provider = font.split('_')[0];
    switch (provider) {
        case 'Google':
            return parseGoogleFont(font);
        default:
            return null;
    }
}

function parseGoogleFont(font: string) {
    const split = font.split('_');
    const family = split[1];
    let formattedFont = '';
    let weight = split[2];

    if (split.length === 2) {
        formattedFont += `${family}|`;
    } else if (split.length > 2) {
        weight = weight.split(',')[0];
        formattedFont += `${family}:${weight}|`;
    }

    return `//fonts.googleapis.com/css?family=${formattedFont}`;
}
