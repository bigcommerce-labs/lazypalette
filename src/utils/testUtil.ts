export const sel = (id: string) => `[data-test-id="${id}"]`;
