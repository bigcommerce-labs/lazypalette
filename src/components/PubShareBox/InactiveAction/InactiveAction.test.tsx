import { mount, shallow } from 'enzyme';
import React from 'react';

import { sel } from '../../../utils/testUtil';

import InactiveAction from './InactiveAction';

describe('InactiveAction', () => {

    const publishButton = sel('publish');
    const useAsActiveButton = sel('use-as-active');

    const mockHandlePublish = jest.fn();
    const mockHandleSave = jest.fn();

    const InactiveActionElement = <InactiveAction
        canPublish={false}
        canSave={false}
        isPrelaunchStore={false}
        handlePublish={mockHandlePublish}
        handleSave={mockHandleSave}
    />;

    const inactiveActionShallow = shallow(InactiveActionElement);
    const inactiveActionMount = mount(InactiveActionElement);

    it('renders', () => {
        expect(inactiveActionShallow).toMatchSnapshot();
    });

    describe('when isPrelaunchStore is true', () => {
        it('should render "Use as Active" button', () => {
            inactiveActionMount.setProps({isPrelaunchStore: true});
            expect(inactiveActionMount.find(useAsActiveButton).hostNodes().length).toBe(1);
            expect(inactiveActionMount.find(publishButton).hostNodes().length).toBe(0);
        });
    });

    describe('when isPrelaunchStore is false', () => {
        it('should render Publish button', () => {
            inactiveActionMount.setProps({isPrelaunchStore: false});
            expect(inactiveActionMount.find(useAsActiveButton).hostNodes().length).toBe(0);
            expect(inactiveActionMount.find(publishButton).hostNodes().length).toBe(1);
        });
    });
});
