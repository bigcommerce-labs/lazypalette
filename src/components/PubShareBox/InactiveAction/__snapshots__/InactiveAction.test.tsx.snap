// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InactiveAction renders 1`] = `
<Fragment>
  <ButtonInput
    disabled={true}
    onClick={[MockFunction]}
    testId="save"
    type="button"
  >
    Save
  </ButtonInput>
  <ButtonInput
    classType="primary"
    disabled={true}
    onClick={[MockFunction]}
    testId="publish"
  >
    Publish
  </ButtonInput>
</Fragment>
`;
